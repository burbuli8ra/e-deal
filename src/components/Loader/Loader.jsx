import React from 'react';
import Styled from './Loader.style';

const Loader = () => {
  return (
    <Styled.Loader data-testid="loader">
      Loading...
    </Styled.Loader>
  );
};

export default Loader;