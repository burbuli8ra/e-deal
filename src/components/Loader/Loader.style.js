import { keyframes } from '@emotion/core';
import styled from '@emotion/styled';
import { colors } from 'theme';

const spin = keyframes`
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
`;

const Loader = styled.div`
  color: ${colors.darkPastelBlue};
  display: flex;
  align-items: center;
  flex-direction: column;
  justify-content: center;
  height: 100%;
  &::before {
    animation: ${spin} 1.4s linear infinite;
    content: '';
    border: 4px solid transparent;
    border-radius: 50%;
    border-top: 4px solid ${colors.darkPastelBlue};
    display: inline-flex;
    height: 24px;
    width: 24px;
    margin-bottom: 6px;
  }
`;
Loader.displayName = 'Loader';


export default {
  Loader
};