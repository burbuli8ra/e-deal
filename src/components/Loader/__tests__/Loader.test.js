import React from 'react';
import { render } from '@testing-library/react';
import Loader from '../Loader';

const setup = props => <Loader {...props} />;

describe('Loader component', () => {
  let props;
  let wrapper;

  beforeEach(() => {
    props = {};
    wrapper = render(setup(props));
  });

  test('should match snapshot', () => {
    const { asFragment } = wrapper;
    expect(asFragment()).toMatchSnapshot();
  });
});