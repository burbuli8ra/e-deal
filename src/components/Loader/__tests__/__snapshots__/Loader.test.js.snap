// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-1ysydv3"
    data-testid="loader"
  >
    Loading...
  </div>
</DocumentFragment>
`;
